src/core-layers/geojson-layer/geojson.js

```diff
@@ -39,25 +38,23 @@ export function getGeojsonFeatures(geojson) {
     return geojson;
   }
 
-  const type = get(geojson, 'type');
-  switch (type) {
-    case 'Point':
-    case 'MultiPoint':
-    case 'LineString':
-    case 'MultiLineString':
-    case 'Polygon':
-    case 'MultiPolygon':
+  assert(geojson.type, 'GeoJSON does not have type');
+
+  switch (geojson.type) {
     case 'GeometryCollection':
-      // Wrap the geometry object in a 'Feature' object and wrap in an array
-      return [{type: 'Feature', properties: {}, geometry: geojson}];
+      assert(Array.isArray(geojson.geometries), 'GeoJSON does not have geometries');
```

Comment:
Super nit: should we specify we are expecting an `Array` type also ? may be the message should be `GeoJSON does not have geometries Array` ? Same for other asserts.